src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/service/haflow/HaFlowUpdateServiceTest.java

```diff
@@ -0,0 +1,456 @@
+import org.openkilda.messaging.command.haflow.HaFlowRequest.Type;
+import org.openkilda.messaging.command.haflow.HaSubFlowDto;
+import org.openkilda.messaging.command.haflow.HaSubFlowPartialUpdateDto;
+import org.openkilda.messaging.command.haflow.HaSubFlowPartialUpdateDto.HaSubFlowPartialUpdateDtoBuilder;
+import org.openkilda.messaging.command.yflow.FlowPartialUpdateEndpoint;
+import org.openkilda.messaging.command.yflow.FlowPartialUpdateEndpoint.FlowPartialUpdateEndpointBuilder;
+import org.openkilda.model.FlowStatus;
+import org.openkilda.model.HaFlow;
+import org.openkilda.model.HaSubFlow;
+import org.openkilda.model.PathComputationStrategy;
+import org.openkilda.model.Switch;
+import org.openkilda.wfm.topology.flowhs.mapper.HaFlowMapper;
+import org.openkilda.wfm.topology.flowhs.service.AbstractHaFlowTest;
+import org.openkilda.wfm.topology.flowhs.service.FlowGenericCarrier;
+
+import com.google.common.collect.Sets;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+import java.time.Instant;
+import java.util.ArrayList;
+
+public class HaFlowUpdateServiceTest extends AbstractHaFlowTest<FlowSegmentRequest> {
+    private static final Switch SWITCH_1 = buildSwitch(SWITCH_ID_1);
+    private static final Switch SWITCH_2 = buildSwitch(SWITCH_ID_2);
+    private static final Switch SWITCH_3 = buildSwitch(SWITCH_ID_3);
+    private static final HaFlow HA_FLOW = buildHaFlow();
+    private static final HaFlowPartialUpdateRequest PATCH_REQUEST = HaFlowPartialUpdateRequest.builder()
+            .haFlowId(HA_FLOW.getHaFlowId())
+            .sharedEndpoint(new FlowPartialUpdateEndpoint(SWITCH_ID_1, PORT_1, VLAN_1, INNER_VLAN_1))
+            .maximumBandwidth(BANDWIDTH_2)
+            .pathComputationStrategy(LATENCY)
+            .maxLatency(MAX_LATENCY_2)
+            .maxLatencyTier2(MAX_LATENCY_TIER_2_2)
+            .encapsulationType(TRANSIT_VLAN)
+            .ignoreBandwidth(false)
+            .periodicPings(true)
+            .pinned(false)
+            .priority(PRIORITY_2)
+            .strictBandwidth(true)
+            .description(DESCRIPTION_2)
+            .allocateProtectedPath(false)
+            .subFlows(newArrayList(
+                    HaSubFlowPartialUpdateDto.builder()
+                            .flowId(SUB_FLOW_ID_1)
+                            .endpoint(new FlowPartialUpdateEndpoint(SWITCH_ID_2, PORT_2, VLAN_2, INNER_VLAN_2))
+                            .description(DESCRIPTION_1)
+                            .build(),
+                    HaSubFlowPartialUpdateDto.builder()
+                            .flowId(SUB_FLOW_ID_2)
+                            .endpoint(new FlowPartialUpdateEndpoint(SWITCH_ID_3, PORT_3, VLAN_3, INNER_VLAN_3))
+                            .description(DESCRIPTION_2)
+                            .build()))
+            .diverseFlowId(HA_FLOW_ID_2)
+            .build();
+
+
+    @Mock
+    private FlowGenericCarrier carrier;
+    HaFlowUpdateService service;
```

Comment:
could be private. I'd separate fields in different blocks: all mocks, all expected results, and so on.

Reply:
will fix in next PR